src/v/cluster/rm_stm.cc

```diff
@@ -784,6 +784,18 @@ ss::future<result<rm_stm::transaction_set>> rm_stm::get_transactions() {
     co_return ans;
 }
 
+ss::future<> rm_stm::abort_transaction(model::producer_identity pid) {
```

Comment:
This method doesn't fail when a transaction with a given pid doesn't exist or the transaction was already committed or aborted. It may lead to confusion: a user thinks a transaction is aborted while in fact it was committed. We may avoid the confusion if you rename and update it to fail when a tx isn't there.

```(cpp)
mark_expired(pid) {
  return get_tx_lock(pid.get_id())->with([this, pid]() {
      return do_mark_expired(pid)
  });
}

do_mark_expired(pid) {  
 if (!co_await sync(_sync_timeout)) {
      co_return false;
  }
  if (!is_known_session(pid)) {
      co_return false;
  }
  _mem_state.expiration.erase(pid);
  co_await do_try_abort_old_tx(pid);
  co_return true;
}
```

Reply:
Please rename all the upcalls, rest endpoint, commit messages & the PR too